src/BatchRequest.php

```diff
@@ -20,27 +20,25 @@
     /**
      * {@inheritdoc}
      */
-    abstract public function sendRequest(RequestInterface $request, array $options = []);
+    abstract public function sendRequest(RequestInterface $request);
 
     /**
      * {@inheritdoc}
      */
-    public function sendRequests(array $requests, array $options = [])
+    public function sendRequests(array $requests)
     {
         $batchResult    = new BatchResult();
-        $batchException = new BatchException();
-        $batchException->setResult($batchResult);
 
         foreach ($requests as $request) {
             try {
-                $batchResult->addResponse($request, $this->sendRequest($request, $options));
+                $batchResult->addResponse($request, $this->sendRequest($request));
```

Comment:
Shouldn't we extract the send part? In a backtrace it might be easier to read where the problem exactly happened. Maybe setting it in a response variable?